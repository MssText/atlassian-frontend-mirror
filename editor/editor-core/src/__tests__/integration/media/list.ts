import { BrowserTestCase } from '@atlaskit/webdriver-runner/runner';
import { getDocFromElement, editable, insertMedia } from '../_helpers';
import {
  goToEditorTestingWDExample,
  mountEditor,
} from '../../__helpers/testing-example-helpers';
import { Appearance } from '../../visual-regression/_utils';

BrowserTestCase(
  `list: insert a media single inside a bullet list`,
  { skip: [] },
  async (client: any, testName: string) => {
    const page = await goToEditorTestingWDExample(client);
    await mountEditor(page, {
      appearance: Appearance.fullPage,
      media: {
        allowMediaSingle: true,
      },
    });
    await page.click('[aria-label="Lists"]');
    await page.waitForSelector('[aria-label="Bullet list"]');
    await page.click('[aria-label="Bullet list"]');
    await page.type(editable, '*');

    await insertMedia(page);
    expect(await page.isVisible('[data-testid="media-file-card-view"]')).toBe(
      true,
    );

    const doc = await page.$eval(editable, getDocFromElement);
    expect(doc).toMatchCustomDocSnapshot(testName);
  },
);

BrowserTestCase(
  `list: insert a media single inside a numbered list`,
  { skip: [] },
  async (client: any, testName: string) => {
    const page = await goToEditorTestingWDExample(client);
    await mountEditor(page, {
      appearance: Appearance.fullPage,
      media: {
        allowMediaSingle: true,
      },
    });
    await page.click('[aria-label="Lists"]');
    await page.waitForSelector('[aria-label="Numbered list"');
    await page.click('[aria-label="Numbered list"');
    await page.type(editable, '*');

    await insertMedia(page);
    expect(await page.isVisible('[data-testid="media-file-card-view"]')).toBe(
      true,
    );

    const doc = await page.$eval(editable, getDocFromElement);
    expect(doc).toMatchCustomDocSnapshot(testName);
  },
);
