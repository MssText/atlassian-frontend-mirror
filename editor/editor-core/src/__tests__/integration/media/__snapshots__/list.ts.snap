// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`list: insert a media single inside a numbered list 1`] = `
Object {
  "content": Array [
    Object {
      "attrs": Object {
        "order": 1,
      },
      "content": Array [
        Object {
          "content": Array [
            Object {
              "content": Array [
                Object {
                  "text": "*",
                  "type": "text",
                },
              ],
              "type": "paragraph",
            },
            Object {
              "attrs": Object {
                "layout": "center",
                "width": null,
              },
              "content": Array [
                Object {
                  "attrs": Object {
                    "__contextId": "DUMMY-OBJECT-ID",
                    "__displayType": null,
                    "__external": false,
                    "__fileMimeType": "image/jpeg",
                    "__fileName": "example.png",
                    "__fileSize": 756325,
                    "__mediaTraceId": null,
                    "alt": "",
                    "collection": "MediaServicesSample",
                    "height": 636,
                    "id": "1234-5678-abcd-efgh",
                    "occurrenceKey": null,
                    "type": "file",
                    "url": null,
                    "width": 640,
                  },
                  "type": "media",
                },
              ],
              "type": "mediaSingle",
            },
          ],
          "type": "listItem",
        },
      ],
      "type": "orderedList",
    },
  ],
  "type": "doc",
}
`;

exports[`list: insert a media single inside a numbered list with restartNumberedLists 1`] = `
Object {
  "content": Array [
    Object {
      "attrs": Object {
        "order": 1,
      },
      "content": Array [
        Object {
          "content": Array [
            Object {
              "content": Array [
                Object {
                  "text": "*",
                  "type": "text",
                },
              ],
              "type": "paragraph",
            },
            Object {
              "attrs": Object {
                "layout": "center",
                "width": null,
              },
              "content": Array [
                Object {
                  "attrs": Object {
                    "__contextId": "DUMMY-OBJECT-ID",
                    "__displayType": null,
                    "__external": false,
                    "__fileMimeType": "image/jpeg",
                    "__fileName": "example.png",
                    "__fileSize": 756325,
                    "__mediaTraceId": null,
                    "alt": "",
                    "collection": "MediaServicesSample",
                    "height": 636,
                    "id": "1234-5678-abcd-efgh",
                    "occurrenceKey": null,
                    "type": "file",
                    "url": null,
                    "width": 640,
                  },
                  "type": "media",
                },
              ],
              "type": "mediaSingle",
            },
          ],
          "type": "listItem",
        },
      ],
      "type": "orderedList",
    },
  ],
  "type": "doc",
}
`;
